### PR TITLE
Updating the network requirements

### PR DESCRIPTION
**Pull request by sudomateo**
Clarifying the network requirements for Terraform Enterprise.

### CODE REVIEW

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 1]
```diff
@@ -77,12 +77,38 @@ SELINUX=permissive
...
+  * **443**: To access the Terraform Enterprise appplication via HTTPS.
+  * **8800**: To access the installer dashboard.
+  * **9870-9880 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+  * **23000-23100 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+
+#### Egress
+
+Terraform Enterprise connects to the following hostnames for egress connections:
+
+* registry.terraform.io
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]: If we're going to add this, we need to clarify when and why it's required.

Comment (sudomateo) [reply content/source/docs/enterprise/private/preflight-installer.html.md]: I agree. I was thinking about following the sub bullet syntax that's below it but I felt that would be a little cluttered.

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 2]
```diff
@@ -77,12 +77,37 @@ SELINUX=permissive
...
 ### Network Requirements
 
-1. Have the following ports available to the Linux instance:
-  * **22** - to access the instance via SSH from your computer. SSH access to the instance is required for administration and debugging.
-  * **8800** - to access the installer dashboard.
-  * **443** and **80** - to access the TFE app (both ports are needed; HTTP will redirect to HTTPS).
-  * **9870-9880 (inclusive)** - for internal communication on the host and its subnet; not publicly accessible.
-  * **23000-23100 (inclusive)** - for internal communication on the host and its subnet; not publicly accessible.
+The system that is running Terraform Enterprise must meet the following network
+requirements.
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]: I don't think this introductory sentence is needed.

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 3]
```diff
@@ -77,12 +77,37 @@ SELINUX=permissive
...
+
+#### Ingress
+
+Terraform Enterprise listens on the following ports for ingress connections:
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]:
```suggestion
```

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 4]
```diff
@@ -77,12 +77,37 @@ SELINUX=permissive
...
+  * **22**: To access the instance via SSH from your computer. SSH access to the instance is required for administration and debugging.
+  * **80**: To access the Terraform Enterprise appplication via HTTP. This port redirects to port 443 for HTTPS.
+  * **443**: To access the Terraform Enterprise appplication via HTTPS.
+  * **8800**: To access the installer dashboard.
+  * **9870-9880 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+  * **23000-23100 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+
+#### Egress
+
+Terraform Enterprise connects to the following hostnames for egress connections:
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]:
```suggestion
```

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 5]
```diff
@@ -77,12 +77,37 @@ SELINUX=permissive
...
+
+* registry.terraform.io (Required if using Terraform 0.12 and later.)
+* releases.hashicorp.com (Not required if a [custom Terraform bundle](/docs/enterprise/run/index.html#custom-and-community-providers) is supplied.)
+
+Additionally these hostnames are accessed by the Installer product in online mode:
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]:
```suggestion
Additionally, these hostnames are accessed in online mode:
```

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 6]
```diff
@@ -77,12 +77,37 @@ SELINUX=permissive
...
+  * **8800**: To access the installer dashboard.
+  * **9870-9880 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+  * **23000-23100 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+
+#### Egress
+
+Terraform Enterprise connects to the following hostnames for egress connections:
+
+* registry.terraform.io (Required if using Terraform 0.12 and later.)
+* releases.hashicorp.com (Not required if a [custom Terraform bundle](/docs/enterprise/run/index.html#custom-and-community-providers) is supplied.)
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]: This needs to be adjusted a bit.

```
* releases.hashicorp.com
* registry.terraform.io (when using Terraform 0.12 and later)

Access to these hostnames is not required if a [custom Terraform bundle](/docs/enterprise/run/index.html#custom-and-community-providers) is supplied.
```

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 7]
```diff
@@ -77,12 +77,37 @@ SELINUX=permissive
...
+
+Additionally these hostnames are accessed by the Installer product in online mode:
+
+* api.replicated.com
+* get.replicated.com
+* quay.io
+* registry-data.replicated.com
+* registry.replicated.com
+
+#### Miscellaneous
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]:
```suggestion
#### Other configuration
```

File: content/source/docs/enterprise/private/preflight-installer.html.md [hunk 8]
```diff
@@ -77,12 +77,34 @@ SELINUX=permissive
...
+  * **22**: To access the instance via SSH from your computer. SSH access to the instance is required for administration and debugging.
+  * **80**: To access the Terraform Enterprise appplication via HTTP. This port redirects to port 443 for HTTPS.
+  * **443**: To access the Terraform Enterprise appplication via HTTPS.
+  * **8800**: To access the installer dashboard.
+  * **9870-9880 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+  * **23000-23100 (inclusive)**: For internal communication on the host and its subnet; not publicly accessible.
+
+#### Egress
+
+The following hostnames are accessed online mode:
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/preflight-installer.html.md]:
```suggestion
The following hostnames are accessed in online mode:
```

whoops, that might have been my fault - overzealous select

**Pull request merged**